near-sdk-macros/src/core_impl/info_extractor/visitor.rs

```diff
@@ -209,26 +220,26 @@ fn is_view(sig: &Signature) -> bool {
     }
 }
 
-fn parse_return_kind(typ: &Type, handles_result: ResultHandling) -> syn::Result<ReturnKind> {
+fn parse_return_kind(
+    typ: &Type,
+    handles_result: ResultHandling,
+    persist_on_error: bool,
+) -> syn::Result<ReturnKind> {
     match handles_result {
-        ResultHandling::NoCheck => Ok(ReturnKind::HandlesResult(typ.clone())),
+        ResultHandling::NoCheck => Ok(ReturnKind::HandlesResultExplicit(typ.clone())),
         ResultHandling::Check => {
             if !utils::type_is_result(typ) {
                 Err(Error::new(typ.span(), "Function marked with #[handle_result] should return Result<T, E> (where E implements FunctionError). If you're trying to use a type alias for `Result`, try `#[handle_result(aliased)]`."))
             } else {
-                Ok(ReturnKind::HandlesResult(typ.clone()))
+                Ok(ReturnKind::HandlesResultExplicit(typ.clone()))
             }
         }
         ResultHandling::None => {
             if utils::type_is_result(typ) {
-                Err(Error::new(
-                    typ.span(),
-                    "Serializing Result<T, E> has been deprecated. Consider marking your method \
-                    with #[handle_result] if the second generic represents a panicable error or \
-                replacing Result with another two type sum enum otherwise. If you really want \
-                to keep the legacy behavior, mark the method with #[handle_result] and make \
-                it return Result<Result<T, E>, near_sdk::Abort>.",
-                ))
+                Ok(ReturnKind::HandlesResultImplicit(crate::StatusResult {
```

Comment:
I would like to mention this change: from some time using Result without #[handle_result] became deprecated. So introducing it again may break contracts that use that old sdk and move to the new version. Please let me know if this change should be reconsidered.

Reply:
It was initially called #[return_result] instead of #[handle_result] - change [applied](https://github.com/near/near-sdk-rs/pull/754/files)

Here is the first issue about this macro and [comment](https://github.com/near/near-sdk-rs/issues/574#issuecomment-923284947) by [austinabell](https://github.com/austinabell). According to the comment I believe it was decided to use explicit attribute so that it does not accidentally break existing contracts. So the macro was introduced in March 2022 along with forbidding to use Result without macro a bit [later](https://github.com/near/near-sdk-rs/pull/757/files).